src/Microsoft.Azure.WebJobs.Host/Bindings/BindingTemplateExtensions.cs

```diff
@@ -63,6 +63,10 @@ private static void ValidateContractCompatibility(IEnumerable<string> parameterN
             {
                 foreach (string parameterName in parameterNames)
                 {
+                    if (string.Equals(parameterName, SysBindingData.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
No. This validation is on public names. No user expression will refer to the internal name. 

---
In reply to: [116887647](https://github.com/Azure/azure-webjobs-sdk/pull/1148#discussion_r116887647) [](ancestors = 116887647)